File: packages/public-api/src/CoordinatesService.ts
import Fetch from "./Fetch.js";
import { handleResponse } from "./handleResponse.js";
import { Response, Problem, ListResponse } from "./Types.js";

export default class CoordinatesService {
  private readonly path = `coordinates`;
  constructor(private readonly fetch: Fetch) {}

  /** Adds a Coordinates field, enabling geographical queries. */
  async add(account_name: string, list_name: string): Promise<Response<ListResponse, Problem>> {
    const path = `${this.path}/${encodeURIComponent(account_name)}/${encodeURIComponent(list_name)}`;
    const response = await this.fetch.post(path, {});
    return handleResponse(response, async () => (await response.json()) as ListResponse);
  }

  /** Removes the Coordinates field. Breaking change - automatically updates the List version. */
  async remove(account_name: string, list_name: string): Promise<Response<ListResponse, Problem>> {
    const path = `${this.path}/${encodeURIComponent(account_name)}/${encodeURIComponent(list_name)}`;
    const response = await this.fetch.delete(path);
    return handleResponse(response, async () => (await response.json()) as ListResponse);
  }
}

File: dotnet/SearchStack.PublicApi/Services/CoordinatesService.cs
using System.Threading;
using System.Threading.Tasks;
using static SearchStack.PublicApi.Url;

namespace SearchStack.PublicApi;

/// <summary>Coordinates field add / remove operations.</summary>
public sealed class CoordinatesService
{
    private const string Path = "coordinates";
    private readonly Transport _transport;

    internal CoordinatesService(Transport transport) => _transport = transport;

    /// <summary>Adds a Coordinates field, enabling geographical queries.</summary>
    public Task<Response<ListResponse>> AddAsync(string accountName, string listName, CancellationToken cancellationToken = default) =>
        _transport.PostAsync<ListResponse>($"{Path}/{Enc(accountName)}/{Enc(listName)}", new object(), cancellationToken);

    /// <summary>Removes the Coordinates field. Breaking change - automatically updates the List version.</summary>
    public Task<Response<ListResponse>> RemoveAsync(string accountName, string listName, CancellationToken cancellationToken = default) =>
        _transport.DeleteAsync<ListResponse>($"{Path}/{Enc(accountName)}/{Enc(listName)}", cancellationToken);
}
